app/src/components/connect/AddSession.tsx

```diff
@@ -32,18 +45,25 @@ const AddSession: React.FC<Props> = ({ primary }) => {
   const { sessionStore } = useStore();
 
   const [label, setLabel] = useState('');
+  const [permissions, setPermissions] = useState('admin');
   const [editing, setEditing] = useState(false);
 
   const toggleEditing = useCallback(() => setEditing(e => !e), []);
   const handleSubmit = useCallback(async () => {
-    const session = await sessionStore.addSession(label, MAX_DATE);
+    const sessionType =
+      permissions === 'admin'
+        ? LIT.SessionType.TYPE_MACAROON_ADMIN
+        : LIT.SessionType.TYPE_MACAROON_READONLY;
+
+    const session = await sessionStore.addSession(label, sessionType, MAX_DATE);
+
     if (session) {
       setLabel('');
       setEditing(false);
     }
   }, [label]);
```

Comment:
I ran into a small issue. First I created an Admin session followed by a Read Only session without reloading the page. The second session was created but with the permissions set to Admin. 

The problem is that this `useCallback` is memoized based on the values passed into the dependencies array. You'll need to add `permissions` into the array to resolve the problem.